### ISSUE TITLE
"Duration" undefined in case of looping AudioBufferSourceNode

### ISSUE THREAD

**Issue by cwilso** [WebAudio/web-audio-api#421]
It's not clear what should happen if you pass a duration to start() with looping enabled.

I would propose, in effect, that duration is functionally equivalent to doing start(start, offset) followed by stop(start+duration).  This mean many possible repetitions of the loop could be played prior to playback being ceased.

Second option would be duration is explicitly ignored if looping is set.

**Comment by joeberkovitz** [comment]
Note the the semantics of `offset` w/r/t computed playback rate are unclear -- see #560 .

**Comment by joeberkovitz** [comment]
We are still going with option 1 above (duration applies to de facto sound output duration not teh buffer content duration) and needs to be documented.

**Comment by kosso** [comment]
Currently there appears to be no way to loop a *segment* (ie: with `offset` *and* `duration` *and* `loop`) of an AudioBuffer. 

This is how I'm trying to do it:

```
...
source = context.createBufferSource();
source.loop = true; 
source.buffer = total_buffer;
source.connect(context.destination);

// Start 'now' at offset seconds for duration seconds.. 
source.start(context.currentTime, offset, duration);

// Only plays once. 
// Expected behaviour: Looping audio starting 'offset' seconds into the buffer with a loop duration of 'duration'.
```

Am I missing something obvious here? Thanks.

**Comment by padenot** [comment]
Use `loopStart` and `loopEnd`.

Please use StackOverflow or the like for questions that are not directly related to the specification document (published at <https://webaudio.github.io/web-audio-api/>), thanks!

**Comment by kosso** [comment]
But surely specifying `loop`as  true, an `offset` to start from and a `duration` to be played,  implicitly defines a `loopStart` and `loopEnd`?

This is a confusing spec.

**Comment by padenot** [comment]
No it's not, it's very well defined. This is an API that is specified, and there is no room for "implicit" definitions.

The definition for the `start` method is [here](https://webaudio.github.io/web-audio-api/#widl-AudioBufferSourceNode-start-void-double-when-double-offset-double-duration), the definition for [`loopStart`](https://webaudio.github.io/web-audio-api/#widl-AudioBufferSourceNode-loopStart) and [`loopEnd`](https://webaudio.github.io/web-audio-api/#widl-AudioBufferSourceNode-loopEnd) are available as well. The overall rendering algorithm for this `AudioNode` is [here](https://webaudio.github.io/web-audio-api/#playback-AudioBufferSourceNode) and a specific section about looping is availble [here](https://webaudio.github.io/web-audio-api/#looping-AudioBufferSourceNode).

This is normative text, and might not be the easier to read, but MDN has an article about the [`AudioBufferSourceNode`](https://developer.mozilla.org/en/docs/Web/API/AudioBufferSourceNode), and [`AudioScheduledSourceNode`](https://developer.mozilla.org/en-US/docs/Web/API/AudioScheduledSourceNode), that are more approachable maybe.

**Comment by kosso** [comment]
Thanks. It was through the MDN docs that I eventually wound up here from. 
After spending a couple of days wondering why Firefox was only playing bufferSources from the right speaker in the test/demo code. (Then the 55.0 update fixed it!) Then I saw an old chromium dev thread which appeared to show others with some confusion over the way looping an ABSN should work. So then I came directly here. 

I did finally get a *segment* of an AudioBuffer finally looping with `loopStart` and `loopEnd` set. 
But it *only* worked when I also set the `offset` *but NOT* the `duration`. 

Thank you for your time.
